From 6d90417f663d5acf3833a392c54be32a5f6e708e Mon Sep 17 00:00:00 2001
From: Praveen Talari <praveen.talari@oss.qualcomm.com>
Date: Fri, 6 Feb 2026 15:16:03 +0530
Subject: [PATCH] spi: qcom-geni: Abort active transfer during system suspend
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Abort any in-progress SPI transfer before invoking the controller’s
suspend routine to ensure a clean and safe suspend state.

Change-Id: If61ef5151850c2989fb485da1a9147326d5a7dc6
Signed-off-by: Praveen Talari <praveen.talari@oss.qualcomm.com>
---
 drivers/spi/spi-geni-qcom.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/spi/spi-geni-qcom.c b/drivers/spi/spi-geni-qcom.c
index 3d17c017b8ea..9847d9727b0d 100644
--- a/drivers/spi/spi-geni-qcom.c
+++ b/drivers/spi/spi-geni-qcom.c
@@ -1532,6 +1532,12 @@ static int __maybe_unused spi_geni_suspend(struct device *dev)
 	int ret;
 
 	trace_spi_info(dev, __func__, "start");
+
+	if (spi->target) {
+		dev_err(dev, "%s: Execute spi_geni_target_abort()\n", __func__);
+		spi_geni_target_abort(spi);
+	}
+
 	ret = spi_controller_suspend(spi);
 	if (ret)
 		return ret;
-- 
2.34.1

